lib/galaxy/model/dataset_collections/types/sample_sheet_workbook.py

```diff
@@ -0,0 +1,622 @@
+)
+
+WorkbookContentField: Base64StringT = Field(
+    ...,
+    title="Workbook Content (Base 64 encoded)",
+    description="The workbook content (the contents of the xlsx file) that have been base64 encoded.",
+)
+PrefixRowsField: Optional[PrefixRowValuesT] = Field(
+    None,
+    title="Prefix sample sheet values",
+    description="An area to pre-populate URIs, etc...",
+)
+
+SampleSheetCollectionType = Literal[
+    "sample_sheet", "sample_sheet:paired", "sample_sheet:paired_or_unpaired", "sample_sheet:record"
+]
+
+ParsedRow = Dict[str, SampleSheetColumnValueT]
+ParsedRows = List[ParsedRow]
+
+AnyLogMessage = Union[InferredColumnMapping, ContentTypeMessage, CsvDialectInferenceMessage]
+
+SampleSheetParseLog = List[AnyLogMessage]
+
+
+class ParsedWorkbook(BaseModel):
+    rows: ParsedRows
+    # extra columns contained in the supplied workbook that have relevant Galaxy metadata
+    # maybe should be thought of as "suffix_columns" since they are after the prefix columns
+    # and user-defined columns.
+    extra_columns: List[ParsedColumn]
+    parse_log: SampleSheetParseLog
+
+
+class CreateWorkbookFromBase64(BaseModel):
+    title: str = CreateTitleField
+    collection_type: SampleSheetCollectionType
+    prefix_columns_type: Literal["URI"] = "URI"
+    column_definitions: Base64StringT
+    prefix_values: Optional[Base64StringT] = None
+
+
+@dataclass
+class CreateWorkbookFromBase64ForCollection:
+    title: str
+    dataset_collection: AnyDatasetCollection
+    column_definitions: Base64StringT
+
+
+class CreateWorkbook(BaseModel):
+    title: str = CreateTitleField
+    collection_type: SampleSheetCollectionType
+    prefix_columns_type: Literal["URI", "ModelObjects"] = "URI"
+    column_definitions: List[SampleSheetColumnDefinitionModel] = ColumnDefinitionsField
+    prefix_values: Optional[InternalPrefixRowValuesT] = None
+
+
+@dataclass
+class CreateWorkbookForCollection:
+    title: str
+    dataset_collection: AnyDatasetCollection
+    column_definitions: List[SampleSheetColumnDefinitionModel] = ColumnDefinitionsField
+
+
+class ParseWorkbook(BaseModel):
+    collection_type: SampleSheetCollectionType
+    prefix_columns_type: Literal["URI", "ModelObjects"] = "URI"
+    column_definitions: List[SampleSheetColumnDefinitionModel] = ColumnDefinitionsField
+    content: str = WorkbookContentField
+
+
+@dataclass
+class ParseWorkbookForCollection:
+    dataset_collection: AnyDatasetCollection
+    column_definitions: List[SampleSheetColumnDefinitionModel] = ColumnDefinitionsField
+    content: str = WorkbookContentField
+
+
+AnyParseWorkbook = Union[ParseWorkbook, ParseWorkbookForCollection]
+
+
+INSTRUCTIONS = [
+    "Use this spreadsheet to describe your samples. For each sample (i.e. each file), ensure all the labeled columns are specified and correct.",
+    "If you're using Google Sheets, data validation will be applied automatically - just make sure no cell values have a red mark indicating they are invalid.",
```

Comment:
how do I do this ? I haven't seen that option ?
<img width="1538" height="676" alt="Screenshot 2025-07-17 at 18 01 27" src="https://github.com/user-attachments/assets/bb5ab60d-ed73-4382-b92c-38020d190664" />

element identifiers are required and accession are non-optional.